solidity/contracts/bridge/RebateStaking.sol

```diff
@@ -0,0 +1,339 @@
+    // planned upgrades of the Bridge contract. If more entires are added to
+    // the struct in the upcoming versions we need to reduce the array size.
+    // See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+    // slither-disable-next-line unused-state
+    uint256[50] __gap;
+
+    event RollingWindowUpdated(uint256 rollingWindow);
+    event UnstakingPeriodUpdated(uint256 unstakingPeriod);
+    event RebatePerTokenUpdated(uint256 rebatePerToken);
+    event RebateReceived(address staker, uint64 rebate);
+    event RebateCanceled(address staker, uint256 requestedAt);
+    event Staked(address staker, uint256 amount);
+    event UnstakeStarted(address staker, uint256 amount);
+    event UnstakeFinished(address staker, uint256 amount);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    modifier onlyBridge() {
+        require(msg.sender == address(bridge), "Caller is not the bridge");
+        _;
+    }
+
+    /// @dev Initializes upgradable contract on deployment.
+    function initialize(
+        address _bridge,
+        address _token,
+        uint256 _rollingWindow,
+        uint256 _unstakingPeriod,
+        uint256 _rebatePerToken
+    ) external initializer {
+        require(
+            _bridge != address(0) && 
+            _token != address(0) && 
+            _rollingWindow != 0, 
+            "Parameters cannot be zero"
+        );
+        bridge = _bridge;
+        token = IERC20Upgradeable(_token);
+        rollingWindow = _rollingWindow;
+        unstakingPeriod = _unstakingPeriod;
+        rebatePerToken = _rebatePerToken;
+
+        __Ownable_init();
+    }
+
+    /// @notice Updates the rolling window.
+    /// @param _newRollingWindow Duration of the rolling window.
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner,
+    ///      - The new rolling window cannot be zero
+    function updateRollingWindow(uint256 _newRollingWindow) external onlyOwner {
+        require(
+            _newRollingWindow != 0, 
+            "Rolling window cannot be zero"
+        );
+        rollingWindow = _newRollingWindow;
+        emit RollingWindowUpdated(rollingWindow);
+    }
+
+    /// @notice Updates the unstaking period.
+    /// @param _newUnstakingPeriod Duration of the unstaking period.
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner
+    function updateUnstakingPeriod(uint256 _newUnstakingPeriod) external onlyOwner {
+        unstakingPeriod = _newUnstakingPeriod;
+        emit UnstakingPeriodUpdated(unstakingPeriod);
+
+    }
+
+    /// @notice Updates the rebate per token.
+    /// @param _newRebatePerToken Rebate coefficient.
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner
+    function updateRebatePerToken(uint256 _newRebatePerToken) external onlyOwner {
+        rebatePerToken = _newRebatePerToken;
```

Comment:
Is `0` a legal value for `rebatePerToken`?

Reply:
I think yes, basically it means that contract is paused (no rebates)